### ISSUE TITLE
`misc-const-correctness` not reported for C code

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#56958]
```c
extern int cb();

void f1()
{
    int res = cb();
    if (res) {}
}
```
https://godbolt.org/z/Tzrb1Yvxq

The documentation refers to C++ coding standards but there's no mention that the check will only be performed on C++ code.

**Comment by njames93** [comment]
@JonasToth I feel like a quick fix is to just update the documentation to say this doesn't run on `C` code. Then we can look into adapting to make it work for `C` code, which I imagine is just a case of adding test cases and removing the language restriction.

**Comment by JonasToth** [comment]
The language restriction was added during review. `const` works different enough in `C` that it was removed. 
I will provide a documentation update as known limitation

**Comment by JonasToth** [comment]
@njames93 i included the limitation in this patch: https://reviews.llvm.org/D132244
That is the separation of the documentation changes done for #56749
